core/src/main/java/hivemall/ftvec/OnehotEncodingUDAF.java

```diff
@@ -0,0 +1,318 @@
+                }
+            } else {
+                listOI = (ListObjectInspector)parameters[0];
+            }
+
+            if (m == Mode.PARTIAL1 || m == Mode.PARTIAL2) {
+                return ObjectInspectorFactory.getStandardListObjectInspector(
+                        PrimitiveObjectInspectorFactory.writableStringObjectInspector);
+            } else {
+                for (int i = 0; i < numFeatures; i++) {
+                    fieldNames.add("feature" + String.valueOf(i));
+                }
+                fieldNames.add("encode");
+                for (int i = 0; i < numFeatures; i++) {
+                    fieldOIs.add(PrimitiveObjectInspectorFactory.writableStringObjectInspector);
+                }
+                fieldOIs.add(ObjectInspectorFactory.getStandardListObjectInspector(
+                    PrimitiveObjectInspectorFactory.writableStringObjectInspector
+                ));
+
+                return ObjectInspectorFactory.getStandardListObjectInspector(
+                    ObjectInspectorFactory.getStandardStructObjectInspector(fieldNames, fieldOIs)
+                );
+            }
+        }
+
+        @Override
+        public AggregationBuffer getNewAggregationBuffer() throws HiveException {
+            EncodingBuffer buf = new EncodingBuffer(numFeatures);
+            reset(buf);
+            return buf;
+        }
+
+        @Override
+        public void reset(AggregationBuffer aggregationBuffer) throws HiveException {
+            EncodingBuffer buf = (EncodingBuffer)aggregationBuffer;
+            buf.reset();
+        }
+
+        @Override
+        public void iterate(AggregationBuffer aggregationBuffer, Object[] parameters) throws HiveException {
+            assert(parameters.length == numFeatures);
+            for (int i = 0; i < numFeatures; i++) {
+                if (parameters[i] == null) {
+                    return;
+                }
+            }
+
+            EncodingBuffer buf = (EncodingBuffer)aggregationBuffer;
+            ComparableList<Text> featuresList = new ComparableList<Text>();
+            for (int i = 0; i < numFeatures; i++) {
+                String s = PrimitiveObjectInspectorUtils.getString(parameters[i], featureOI.get(i));
+                featuresList.add(new Text(s));
+            }
+            buf.add(featuresList);
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Object terminatePartial(AggregationBuffer aggregationBuffer) throws HiveException {
+            EncodingBuffer buf = (EncodingBuffer)aggregationBuffer;
+            List<Text> ret = new ArrayList<Text>();
+            // TODO: HIVE-11176 - Fail to passing List<Text> due to class cast exception.
+            // As workaround, string type joined under score is passed.
+            for (ComparableList<Text> features : buf.getFeaturesSet()) {
+                ret.add(new Text(StringUtils.join(features, "_")));
```

Comment:
What happens if feature contains `_` on `merge()`. It would produce invalid splits.

I think it is better to have `byte[] EncodingBuffer#serialize()` method and  static`EncodingBuffer.decode(byte[] b)` method.